koffeebreak/history_manager.py

```diff
@@ -1,31 +1,209 @@
+                        time['all'] += delta
+                        time['work'] += delta
+                    previous_state = message
+                    start_timer = message_time
+                    long_break['count'] += 1
+                elif message == 'postpone break':
+                    if previous_state == 'start short break':
+                        short_break['count'] -= 1
+                        short_break['postponed'] += 1
+                    elif previous_state == 'start long break':
+                        long_break['count'] -= 1
+                        long_break['postponed'] += 1
+                elif message == 'skip break':
+                    if previous_state == 'start short break':
+                        short_break['count'] -= 1
+                        short_break['skipped'] += 1
+                    elif previous_state == 'start long break':
+                        long_break['count'] -= 1
+                        long_break['skipped'] += 1
+                elif message == 'break at the computer':
+                    if previous_state == 'start short break':
+                        short_break['at the computer'] += 1
+                    elif previous_state == 'start long break':
+                        long_break['at the computer'] += 1
+                elif message == 'pause program':
+                    if previous_state == 'start work':
+                        delta = message_time - start_timer
+                        time['all'] += delta
+                        time['work'] += delta
+                    elif previous_state == 'start short break':
+                        delta = message_time - start_timer
+                        time['all'] += delta
+                        time['short break'] += delta
+                    elif previous_state == 'start long break':
+                        delta = message_time - start_timer
+                        time['all'] += delta
+                        time['long break'] += delta
+                elif message == 'resume program':
+                    start_timer = message_time
+                elif message == 'close program':
+                    if previous_state == 'start work':
+                        delta = message_time - start_timer
+                        time['all'] += delta
+                        time['work'] += delta
+                    if previous_state == 'start short break':
+                        delta = message_time - start_timer
+                        time['all'] += delta
+                        time['short break'] += delta
+                    if previous_state == 'start long break':
+                        delta = message_time - start_timer
+                        time['all'] += delta
+                        time['long break'] += delta
+
+        # add the difference between current time and the last time in history
+        delta = datetime.strptime(datetime.now().strftime('%Y-%m-%d %H:%M:%S'),
```

Comment:
Is it a joke?:) 
`datetime.strptime(datetime.now().strftime('%Y-%m-%d %H:%M:%S')`
is just 
`datetime.now()`